distributed/client.py

```diff
@@ -394,6 +396,16 @@ def __init__(self, address=None, start=True, loop=None, timeout=5,
         self.connection_args = self.security.get_connection_args('client')
         self._connecting_to_scheduler = False
 
+        if loop is None:
+            self._should_close_loop = None
+            if in_ioloop():
+                self.loop = IOLoop.current()
```

Comment:
> I'd like to push on if possible. Also happy for other people to take this on.

Fair enough. I'll try that next week.

> I'm curious about your thoughts on the explicit loop argument. Can you explain more about how it is useless and fragile in our case?

As for useless, I don't see what use case it solves: for sanity's sake, you always want a distinct IO loop per thread, and that's already Tornado's model (if you use `IOLoop.current(instance=False) or IOLoop()`, for example).

As for dangerous, people may well pass the wrong loop if they don't understand what we're doing with it. Not to mention it makes our loop-handling code more complex.